### PR TITLE
Add Model Classes, Associations, Data Insertion, and Custom Methods

### PR DESCRIPTION
**Pull request by Zafron047**
## Dear Reviewer,

This pull request addresses the project requirements by adding model classes for all entities, setting up associations between models, using models to insert data, and creating custom methods as specified in the project guidelines.

### Changes Made following ERD Diagram:
blog_app_erd_v1_1

- Created model classes for all entities based on the provided ERD diagram.
- Established associations between models, including setting `author_id` as a foreign key for the `users` table in the `posts` model.
- Utilized the 'rails c' command to open the Rails console and inserted sample data as required.
- Created custom methods in the models:
    - Users: Implemented a method that returns the 3 most recent posts for a given user.
    - Posts: Added methods to update the posts counter for a user and return the 5 most recent comments for a given post.
    - Comments: Implemented a method to update the comments counter for a post.
    - Likes: Added a method to update the likes counter for a post.
- Ensured that all custom methods are tested and working as expected in the Rails console.

## Screenshot:
1.  Create at least one user by running the following code:
```ruby
   first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
   second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
``` 

2. Create at least 4 posts written by one of the users you created by running the following code:
```ruby
   first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
``` 

3. ### In this I have tested this below code block with multiple posts to ensure it works with any post.
```ruby
   Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
``` 

### Checklist

- [x] Created model classes for all entities.
- [x] Set up associations between models.
- [x] Inserted sample data using the 'rails c' command.
- [x] Implemented custom methods in the models.
- [x] Tested and verified the functionality of custom methods in the Rails console.

Please review this pull request and provide feedback or approve to merge this branch.

Regards,

@Zafron047

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,2 +1,8 @@
 class Like < ApplicationRecord
+  belongs_to :post, class_name: 'Post', foreign_key: :post_id
+  belongs_to :user, class_name: 'User', foreign_key: :user_id
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (mckent05) [inline app/models/like.rb:10]: - You have done well so far, but just as you did for the `post` and `comment` model, please ensure the likes_counter is updated when a model is created and saved in the database. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Zafron047, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job creating and setting up associations between models 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (Zafron047) [reply app/models/like.rb:10]: Thanks a lot @mckent05 for identifying the missing method call.

I have now updated the class:
Update_like_counter

**Comment by Zafron047** [comment]
@mckent05 I have made the necessary changes as you required:

Update_like_counter

**Review by KDlamini — APPROVED** [review]
### Hi @Zafron047 🙋‍♀️,  

Excellent work! 💯 
You have done a great job processing data in models for the blog app. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞

✔️  Good PR title and description
✔️  Well-documented and professional README 
✔️  `User`, `Post`, `Comment`, and `Like` models are present inside `app/models/` directory
✔️  All required methods are created

# Status: Approved ✔️ 

LeonardoInGatsby

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Zafron047** [comment]
Thanks @KDlamini for the approval.

**Pull request merged**